core/server-entry.ts

```diff
@@ -56,28 +58,26 @@ function getHostFromHeader (headers: string[]): string {
 }
 
 export default async context => {
-  const { app, router, store, initialState } = await createApp(context, context.vs && context.vs.config ? context.vs.config : buildTimeConfig)
+  let storeCode = context.vs.storeCode
+  if (config.storeViews.multistore === true) {
+    if (!storeCode) { // this is from url
+      const currentRoute = Object.assign({ path: queryString.parseUrl(context.url).url/* this gets just the url path part */, host: context.server.request.headers.host })
+      storeCode = storeCodeFromRoute(currentRoute)
+    }
+  }
+  const { app, router, store, initialState } = await createApp(context, context.vs && context.vs.config ? context.vs.config : buildTimeConfig, storeCode)
+
+  RouterManager.flushRouteQueue(router)
```

Comment:
we should move it to `createApp` to avoid the risk that this line will be missed across the upgrades/user stores

Reply:
That's what I thought at first, but the client has to `registerDynamicRoutes` between createApp and flushRouteQueue